src/item/HangingSign.php

```diff
@@ -40,13 +44,26 @@ public function __construct(
 		parent::__construct($identifier, $name);
 	}
 
-	public function getPlacementBlock(Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector) : Block{
+	public function getPlacementBlock(?Player $player, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector) : Block{
 		//we don't verify valid placement conditions here, only decide which block to return
-		$result = $face === Facing::DOWN ?
-			$blockReplace->getSide(Facing::UP)->getSupportType(Facing::DOWN) === SupportType::CENTER ?
-				$this->centerPointCeilingVariant :
-				$this->edgePointCeilingVariant
-			: $this->wallVariant;
+		if($face === Facing::DOWN){
+			if($player !== null && $player->isSneaking()){
+				return clone $this->centerPointCeilingVariant;
+			}
+
+			//we select the center variant when support is edge/wall sign with perpendicular player facing,
+			//support is a center sign itself, or support provides center support.
+			//otherwise use the edge variant.
+			$support = $blockReplace->getSide(Facing::UP);
+			$result =
+				(($support instanceof CeilingEdgesHangingSign || $support instanceof WallHangingSign) && ($player === null || Facing::axis($player->getHorizontalFacing()) !== Facing::axis($support->getFacing()))) ||
+				$support instanceof CeilingCenterHangingSign ||
+				$support->getSupportType(Facing::DOWN) === SupportType::CENTER ?
+					$this->centerPointCeilingVariant :
+					$this->edgePointCeilingVariant;
```

Comment:
I'm thinking this could logically be reduced to:
`$this->edgePointCeilingVariant->canBeSupportedAt($support) ? $this->edgePointCeilingVariant : $this->centerPointCeilingVariant`
Since ceiling center signs have a wider range of acceptable support cases than ceiling edge signs.

If we could publicise that logic or otherwise extract it, we could do that. What do you think?
Also raises a more general question around `StaticSupportTrait->canBeSupportedAt()` on whether it should be public or not. I'm undecided.